okhttp-sse/src/main/kotlin/okhttp3/sse/EventSources.kt

```diff
@@ -26,29 +27,35 @@ object EventSources {
     level = DeprecationLevel.HIDDEN,
   )
   @JvmStatic
-  fun createFactory(client: OkHttpClient) = createFactory(client as Call.Factory)
+  fun createFactory(client: OkHttpClient) = client.asEventSourceFactory()
 
+  @Deprecated(
+    message = "Moved to extension function.",
+    replaceWith =
+      ReplaceWith(
+        expression = "callFactory.asEventSourceFactory()",
+        imports = ["okhttp3.sse.EventSource.Factory.Companion.asEventSourceFactory"],
+      ),
+    level = DeprecationLevel.WARNING,
+  )
   @JvmStatic
   fun createFactory(callFactory: Call.Factory): EventSource.Factory =
-    EventSource.Factory { request, listener ->
-      val actualRequest =
-        if (request.header("Accept") == null) {
-          request.newBuilder().addHeader("Accept", "text/event-stream").build()
-        } else {
-          request
-        }
-
-      RealEventSource(actualRequest, listener).apply {
-        connect(callFactory)
-      }
-    }
+    callFactory.asEventSourceFactory()
 
+  @Deprecated(
+    message = "Moved to extension function.",
+    replaceWith =
+      ReplaceWith(
+        expression = "response.toEventSource(listener)",
```

Comment:
This doesn't seem equivalent.

Reply:
That feels strange. an EventSource is something you can start, and cancel. The Call form enqueues the request and returns immediately.

But this one is not a blocking constructor, which seems unexpected. 

Maybe the current `processResponse` is still the right API here?

Reply:
Isn't it a blocking call? Processing all events on the Response?